packages/core/comparison/conductor-tasks/compareFiles.ts

```diff
@@ -37,24 +54,32 @@ const compareFiles: ConductorWorker = {
         logs.push(res.message)
       } else {
         if (isPass(res.comparisonResult)) {
-          count.pass += 1
+          recordPass(resultRecord, res.phase)
         } else {
-          count.fail += 1
-          logs.push(`Comparison failed: ${res.s3Path}`)
+          recordFail(resultRecord, res.phase)
+          logs.push(`Phase ${res.phase} comparison failed: ${res.s3Path}`)
         }
       }
     })
 
     const nonErrorTestResults = allTestResults.filter((res) => !isError(res)) as ComparisonResult[]
 
-    const recordResultsInDynamoResult = await recordResultsInDynamo(nonErrorTestResults, gateway)
-    if (isError(recordResultsInDynamoResult)) {
-      return failed("Failed to write results to Dynamo")
-    }
+    const phases = [1, 2]
+
+    phases.forEach(async (phase) => {
+      const phaseResults = nonErrorTestResults.filter((res) => res.phase === phase) as ComparisonResult[]
+      const gateway = new DynamoGateway(createDynamoDbConfig(phase))
+      const recordPhaseResults = await recordResultsInDynamo(phaseResults, gateway)
+      if (isError(recordPhaseResults)) {
+        return failed(`Failed to write phase ${phase} results to Dynamo`)
+      }
+    })
 
-    logs.push(`Results of processing: ${count.pass} passed. ${count.fail} failed`)
+    logs.push(
```

Comment:
The majority of the time this will only handle a single message so it might be worth conditionally building the log message because most of the time it will print `undefined` for one of the phases

Reply:
I've added a function to build only relevant logs